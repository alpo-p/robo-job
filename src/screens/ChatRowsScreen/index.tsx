import React, { useState } from 'react'
import { SafeAreaView } from 'react-native-safe-area-context'
import ChatsContainer from './components/ChatRowsContainer'
import SearchBar from './components/SearchBar'
import FilterTopBar from './components/FilterTopBar'

export type StateFilter = 'all' | 'unread' | 'unfinished'

export default () => {
  const [searchFilter, setSearchFilter] = useState('')
  const [stateFilter, setStateFilter] = useState<StateFilter>('all')

  return (
    <SafeAreaView>
      <SearchBar
        searchFilter={searchFilter}
        setSearchFilter={setSearchFilter}
      />
      <FilterTopBar filter={stateFilter} setStateFilter={setStateFilter} />
      <ChatsContainer searchFilter={searchFilter} stateFilter={stateFilter} />
    </SafeAreaView>
  )
}
