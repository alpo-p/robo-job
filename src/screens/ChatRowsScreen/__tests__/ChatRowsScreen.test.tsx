import React from 'react'
import create from '../../../testUtils/rendererCreateWithNavigation'
import ChatRowsScreen from '../index'

describe('ChatRowsScreen', () => {
  it('should match snapshot', () => {
    const tree = create(<ChatRowsScreen />)
    expect(tree).toMatchSnapshot()
  })
})
