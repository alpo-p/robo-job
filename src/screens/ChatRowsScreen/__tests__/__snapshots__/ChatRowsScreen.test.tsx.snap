// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatRowsScreen should match snapshot 1`] = `
<RNCSafeAreaView>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#f1f1f1",
          "borderRadius": 8,
          "flexDirection": "row",
          "height": 32,
          "margin": 16,
        },
        Object {},
      ]
    }
  >
    <Text />
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Search"
      placeholderTextColor="rgba(28, 28, 30, 0.4)"
      rejectResponderTermination={true}
      returnKeyType="done"
      style={
        Object {
          "color": "#000000",
          "width": "70%",
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderBottomColor": "#000000",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "opacity": 1,
          "width": "33%",
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "#000000",
            "fontFamily": "DamascusBold",
            "fontSize": 14,
            "marginBottom": 8,
            "marginRight": 8,
            "textTransform": "none",
          }
        }
      >
        All
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderBottomColor": "#000000",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "opacity": 1,
          "width": "33%",
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "#000000",
            "fontFamily": "Damascus",
            "fontSize": 14,
            "marginBottom": 8,
            "marginRight": 8,
            "textTransform": "none",
          }
        }
      >
        Unread
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderBottomColor": "#000000",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "opacity": 1,
          "width": "33%",
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "#000000",
            "fontFamily": "Damascus",
            "fontSize": 14,
            "marginBottom": 8,
            "marginRight": 8,
            "textTransform": "none",
          }
        }
      >
        Unfinished
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "marginTop": 8,
      }
    }
  />
</RNCSafeAreaView>
`;
